src/middleware.ts

```diff
@@ -0,0 +1,131 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import {Request, Response, NextFunction} from 'express';
+
+const PUBSUB_EVENT_TYPE = 'google.pubsub.topic.publish';
+const PUBSUB_MESSAGE_TYPE =
+  'type.googleapis.com/google.pubsub.v1.PubsubMessage';
+const PUBSUB_SERVICE = 'pubsub.googleapis.com';
+
+interface RawPubSubBody {
+  subscription: string;
+  message: {
+    data: string;
+    messageId: string;
+    attributes: {[key: string]: string};
+  };
+}
+
+interface PubSubEventBody {
```

Comment:
- Can we put interfaces in `src/functions.ts` with our other interfaces?
- Can we partially re-use / extend an existing interface?

Reply:
One bit I am confused about, this interface is a private implementation detail of the logic in this module. My understanding is the stuff in `functions.ts` is that it is public for clients of the framework.

Reply:
For interfaces that developers will use to re-create HTTP POST request event payloads, they should be exported in the `functions.ts`. Otherwise, if private – no user will ever import it – then okay to keep private.

Public interfaces are kept in that file and exposed as stated here: https://github.com/GoogleCloudPlatform/functions-framework-nodejs/blob/master/package.json#L10

Reply:
OK I think it makes sense to keep this one private then.